providers/base/units/watchdog/jobs.pxu

```diff
@@ -1,44 +1,139 @@
-id: watchdog/detect
+id: watchdog/check-timeout
 category_id: com.canonical.plainbox::power-management
-_summary: Detect the presence of a Hardware Watchdog
+_summary: Check the timeout of Hardware Watchdog
+_description:
+    Check the value of RuntimeWatchdogUSec that shouldn't be 0 or unset
+    in OEM images because that means that the watchdog is disabled.
 flags: simple
 imports: from com.canonical.plainbox import manifest
-requires: manifest.has_hardware_watchdog == 'True'
-command: udev_resource.py -f WATCHDOG
+requires:
+    manifest.has_hardware_watchdog == 'True'
+    image_source_and_type.source == 'oem'
+command: watchdog_config_test.py --check-time
+
 
-id: watchdog/systemd-config
-_summary: Check if the hardware watchdog is properly configured
-template-engine: jinja2
-command: watchdog_config_test.py
+id: watchdog/check-service
 category_id: com.canonical.plainbox::power-management
+_summary: Check the watchdog.service is enabled or not
+_description:
+    Check the state of the watchdog.service. It must be disabled/missing 
+    on both classic and core images since 20.04. Else it must be enabled.
 flags: simple
 imports: from com.canonical.plainbox import manifest
-requires: manifest.has_hardware_watchdog == 'True'
+requires:
+    manifest.has_hardware_watchdog == 'True'
+command: watchdog_config_test.py --check-service
 
-id: watchdog/trigger-system-reset-auto
-depends: watchdog/systemd-config
-_summary: Test that the watchdog module can trigger a system reset
+
+id: watchdog/probe-module
+category_id: com.canonical.plainbox::power-management
+_summary: Probe the suitable module for watchdog
+_description:
+    Probe the module 'WATCHDOG_TYPE'.
+    This job can only be execute on the Stock Classic image,
+    because the watchdog module is not probed automatically.
+user: root
+flags: simple
+imports: from com.canonical.plainbox import manifest
+requires:
+    manifest.has_hardware_watchdog == 'True'
+    image_source_and_type.source == 'stock'
+    image_source_and_type.type == 'classic'
+environ: WATCHDOG_TYPE
+command:
+    if [[ -z "$WATCHDOG_TYPE" ]]; then
+        >&2 echo "WATCHDOG_TYPE is not available"
+        exit 1
+    fi
+    echo "Trying to probe '$WATCHDOG_TYPE' module"
+    if ! modprobe "$WATCHDOG_TYPE"; then
+        >&2 echo "Unable to probe the '$WATCHDOG_TYPE' module"
+        exit 1
+    fi
+    if ! lsmod | grep -q -i "$WATCHDOG_TYPE"; then
+        >&2 echo "Unable to find the '$WATCHDOG_TYPE' module after probing it"
+        exit 1
+    fi
+
+
+id: watchdog/detect
+category_id: com.canonical.plainbox::power-management
+_summary: Detect presence of a Hardware Watchdog
+_description:
+    Detect the watchdog is under the /sys/class/watchdog/ path and no other type of watchdog
+flags: simple
+imports: from com.canonical.plainbox import manifest
+requires:
+    manifest.has_hardware_watchdog == 'True'
+environ: WATCHDOG_TYPE WATCHDOG_IDENTITY
```

Comment:
@Hook25 I notice that in my jobs I require some environment variables like WATCHDOG_TYPE and
 WATCHDOG_IDENTITY, however, the machines in the sru pool has not been configured those variables, as a result, it may break the sru testing.
Do you have any idea how we can prevent from this?